python/cuml/utils/memory_utils.py

```diff
@@ -29,6 +29,28 @@
         pass
 
 
+def with_cupy_rmm(func):
+    """
+
+    Decorator to call CuPy functions with RMM memory management. Use it
+    to decorate any function that will call CuPy functions, this will ensure
+    that those calls use RMM for memory allocation instead of the default
+    CuPy pool. Example:
+
+    .. code-block:: python
+
+        @with_cupy_rmm
+        def fx(...):
+            a = cp.arange(10) # uses RMM for allocation
+
+    """
+    def cupy_rmm_wrapper(*args, **kwargs):
```

Comment:
Weird decorator magic, but I think you want to use:
```
from functools import wraps

@wraps(func)
def cupy_rmm_wrapper(...)
```

This ensures that `help(wrapped_func)` doesn't get messed up.

Reply:
Always forget this

Reply:
Looks good